### PR TITLE
Monkey-patching `NodeRole` class to avoid role parsing bug.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This PR is doing some monkey-patching to fix parsing of the `search` role for OpenSearch nodes. This is required due to a lack of support for this in the OpenSearch Java client (bug filed [here](opensearch-project/opensearch-java#893)) and should be removed after the bug was fixed upstream.

Refs #14788.

/nocl

/jpd https://github.com/Graylog2/graylog-plugin-enterprise/pull/6803

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by mpfz0r** [comment]
It doesn't stumble over the search role anymore,  but..

```
org.graylog.shaded.opensearch2.org.opensearch.OpenSearchException: Could not read nodes list from indexer!
	at org.graylog.storage.opensearch2.OpenSearchClient.exceptionFrom(OpenSearchClient.java:254) ~[classes/:?]
	at org.graylog.storage.opensearch2.OpenSearchClient.execute(OpenSearchClient.java:177) ~[classes/:?]
	at org.graylog.storage.opensearch2.ClusterAdapterOS2.nodesInfo(ClusterAdapterOS2.java:305) ~[classes/:?]
	at org.graylog.plugins.archive.job.ArchiveParallelizationAdvisor.getParallelismFactor(ArchiveParallelizationAdvisor.java:41) ~[classes/:?]
	at org.graylog.plugins.archive.job.ArchiveCreateJob.archiveIndex(ArchiveCreateJob.java:261) [classes/:?]
	at org.graylog.plugins.archive.job.ArchiveCreateJob.archiveIndices(ArchiveCreateJob.java:212) [classes/:?]
	at org.graylog.plugins.archive.job.ArchiveCreateJob.run(ArchiveCreateJob.java:169) [classes/:?]
	at org.graylog.plugins.archive.job.ArchiveCreateSystemJob.execute(ArchiveCreateSystemJob.java:35) [classes/:?]
	at org.graylog2.system.jobs.SystemJobManager.executeJob(SystemJobManager.java:116) [classes/:?]
	at org.graylog2.system.jobs.SystemJobManager$1.run(SystemJobManager.java:93) [classes/:?]
	at com.codahale.metrics.InstrumentedScheduledExecutorService$InstrumentedRunnable.run(InstrumentedScheduledExecutorService.java:241) [metrics-core-4.2.25.jar:4.2.25]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) [?:?]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) [?:?]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) [?:?]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.base/java.lang.Thread.run(Thread.java:840) [?:?]
Caused by: org.opensearch.client.util.MissingRequiredPropertyException: Missing required property 'NodeInfoAction.destructiveRequiresName'
	at org.opensearch.client.util.ApiTypeHelper.requireNonNull(ApiTypeHelper.java:89) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.opensearch.nodes.info.NodeInfoAction.<init>(NodeInfoAction.java:57) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.opensearch.nodes.info.NodeInfoAction.<init>(NodeInfoAction.java:50) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.opensearch.nodes.info.NodeInfoAction$Builder.build(NodeInfoAction.java:114) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.opensearch.nodes.info.NodeInfoAction$Builder.build(NodeInfoAction.java:94) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:92) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:55) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectDeserializer$FieldObjectDeserializer.deserialize(ObjectDeserializer.java:81) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:185) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:146) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:87) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:91) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:55) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectDeserializer$FieldObjectDeserializer.deserialize(ObjectDeserializer.java:81) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:185) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:146) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:87) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:91) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:55) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.JsonpDeserializerBase$StringMapDeserializer.deserialize(JsonpDeserializerBase.java:369) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.JsonpDeserializerBase$StringMapDeserializer.deserialize(JsonpDeserializerBase.java:355) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:87) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectDeserializer$FieldObjectDeserializer.deserialize(ObjectDeserializer.java:81) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:185) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:146) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:87) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:91) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:55) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.transport.rest_client.RestClientTransport.decodeResponse(RestClientTransport.java:317) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.transport.rest_client.RestClientTransport.getHighLevelResponse(RestClientTransport.java:283) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.transport.rest_client.RestClientTransport.performRequest(RestClientTransport.java:143) ~[opensearch-java-2.9.1.jar:?]
	at org.opensearch.client.opensearch.nodes.OpenSearchNodesClient.info(OpenSearchNodesClient.java:145) ~[opensearch-java-2.9.1.jar:?]
	at org.graylog.storage.opensearch2.ClusterAdapterOS2.lambda$nodesInfo$19(ClusterAdapterOS2.java:305) ~[classes/:?]
	at org.graylog.storage.opensearch2.OpenSearchClient.execute(OpenSearchClient.java:175) ~[classes/:?]
```

**Comment by thll** [comment]
> It doesn't stumble over the search role anymore, but..

looks like this is a different bug that is already being worked on: https://github.com/Graylog2/graylog2-server/pull/18647

File: graylog-storage-opensearch2/pom.xml [hunk 1]
```diff
@@ -213,6 +213,9 @@
                             <Graylog-Plugin-Properties-Path>${project.groupId}.${project.artifactId}</Graylog-Plugin-Properties-Path>
                         </manifestEntries>
                     </archive>
+                    <archiverConfig>
+                        <duplicateBehavior>add</duplicateBehavior>
+                    </archiverConfig>
```
Comment (bernd) [inline graylog-storage-opensearch2/pom.xml:218]: Let's exclude the duplicate file in the `assembly.xml` descriptor file (adding a comment) so we don't have to reconfigure the archiver. I think that's safer than allowing duplicates.

Comment (dennisoelkers) [reply graylog-storage-opensearch2/pom.xml:218]: I would also prefer this. How would that look, syntax-wise?

Comment (moesterheld) [reply graylog-storage-opensearch2/pom.xml:218]: You can take a look at https://github.com/Graylog2/graylog2-server/pull/18658/commits/e0040b19ae1c321dd062fcb7943aa9f2de23f692 or wait for the merge and add it to the excludes there

Comment (janheise) [reply graylog-storage-opensearch2/pom.xml]: cherry picked @moesterheld changes + removed duplicate behaviour, added class to exclusion

**Review by bernd — APPROVED** [review]

**Pull request merged**